File: src/main/java/org/bk/books/web/dto/AuthorDto.java
package org.bk.books.web.dto;

import org.bk.books.domain.entity.author.Author;

public record AuthorDto(String id, String name, int version) {
  public static AuthorDto map(Author author) {
    return new AuthorDto(author.id().id().toString(), author.name(), author.version());
  }
}

File: src/main/kotlin/org/bk/graphql/web/model/AuthorDto.kt
package org.bk.graphql.web.model

import org.bk.graphql.model.Author

data class AuthorDto(
    val id: String,
    val firstName: String,
    val lastName: String,
    val version: Int

) {
    companion object {
        fun map(author: Author): AuthorDto {
            return AuthorDto(author.id, author.firstName, author.lastName, author.version)
        }
    }
}


